front_end/src/App.js

```diff
@@ -0,0 +1,106 @@
+import Products from './components/Products';
+import { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { fetchProducts } from './actions/productActions';
+import Pagination from './components/Pagination';
+import 'bootstrap';
+import 'bootstrap/dist/css/bootstrap.min.css';
+import 'bootstrap/dist/js/bootstrap.bundle.min.js';
+
+const App = () => {
+  const [price, setPrice] = useState('')
+  const [category, setCategory] = useState('')
+  const [query, setQuery] = useState('')
+  const [currentPage, setCurrentPage] = useState(1)
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchProducts())
+  }, [dispatch])
+
+  const handleCategoryClick = (e) => {
+    e.preventDefault()
+    console.log('Ctg')
+    setCategory(e.target.value)
```

Comment:
The way you have this setup, you aren't able to get value - log e to see what I mean